filer.py

```diff
@@ -16,15 +12,28 @@ def get_filer(self,device=None,tenant=None):
         print("Device not found.")
         sys.exit("Exiting...")
 
-def get_filers(self):
-    """Return all connected Filers from each Tenant"""
+def get_filers(self,all_tenants=False):
+    """Return all connected Filers from Admin Portal or Tenant"""
     connected_filers = []
-    for tenant in self.portals.tenants():
-        self.portals.browse(tenant.name)
-        all_filers = self.devices.filers(include=[
-                'deviceConnectionStatus.connected',
-                'deviceReportedStatus.config.hostname'])
-        for filer in all_filers:
+    if all_tenants is True:
+        self.portals.browse_global_admin()
+        tenant = self.users.session().user.tenant
+        logging.info("Getting all Filers since tenant is " + tenant)
+        for tenant in self.portals.tenants():
+            self.portals.browse(tenant.name)
+            all_filers = self.devices.filers(include=[
+                    'deviceConnectionStatus.connected',
+                    'deviceReportedStatus.config.hostname'])
+            for filer in all_filers:
```

Comment:
You should use list comprehension instead of looping and extend the array.
`connected_filers.extend([filer for filer in all_filers if filer.deviceConnectionStatus.connected])`
The same goes for both the "if" and the "else"

Reply:
Will do. Thanks for example code.